### PR TITLE
[pt2e][quant] Add weight only quantization for xnnpack

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118454

Summary:
with fp32 activations

Test Plan:
python test/test_quantization.py -k test_weight_only_linear_int4_weight

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/quantization/pt2e/test_xnnpack_quantizer.py [hunk 1]
```diff
@@ -582,6 +582,63 @@ def test_dynamic_linear_int4_weight(self):
                 qconfig_mapping,
             )
 
+    def test_weight_only_linear_int4_weight(self):
+        quantizer = XNNPACKQuantizer()
+        quantization_config = get_symmetric_quantization_config(
+            is_per_channel=True,
+            is_weight_only=True,
+            weight_qmin=0,
+            weight_qmax=15,
```
Comment (digantdesai) [inline test/quantization/pt2e/test_xnnpack_quantizer.py:591]:
```suggestion
            weight_qmin=-8,
            weight_qmax=7,
```

**Review by digantdesai — APPROVED** [review]
Looks good. Thanks. I can wire this up at least for fp32 now. It will take more work to do fp16.

File: test/quantization/pt2e/test_xnnpack_quantizer.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/pt2e/test_xnnpack_quantizer.py:591]: I thought xnnpack is using 0, 15? that's why they have zero_point = 8 right?

Comment (digantdesai) [reply test/quantization/pt2e/test_xnnpack_quantizer.py]: This feels like better UX with our symmetric quant theme for xnnpack. We can shift the weights to zp=8 and make then uint8 during the lowering.

**Pull request closed without merging**